### PR TITLE
Use an rgbasm macro instead of a Python preprocessor to output message characters as 16-bit words

### PR DESCRIPTION
**Pull request by Rangi42**
The core of the `text` macro is to iterate over each character of the string it's given with `STRSUB`, and output the individual characters as `dw`s.

This is made somewhat more complicated by the current rgbasm lexer. It handles macro args before it handles string quoting, so `text "ABC, DEF"` is read as two arguments: <`"ABC`> and <` DEF"`>. Thus the macro needs to (a) ignore the quote characters, and (b) restore the comma character between arguments. Ignoring quotes is trivial with `STRCMP`, and it outputs a comma before handling each argument after the first one.

In other words, `text "ABC, DEF; GHI"` would work as expected (although no Japanese strings have comma characters).

When the lexer gets updated, the macro should be simplified.

### CODE REVIEW

**Review by mid-kid — COMMENTED** [review]
The proposed changes look good, however, a possible (rather likely at this point) future change to rgbds will break this macro (https://github.com/rednex/rgbds/pull/557), and as of right now it already does (though the version this feature is based on is a bit dated).
I won't be merging this until either this rgbds change is introduced, rejected for good, or proper widechar support is added, as I want to keep the code with this big a potential to rot to a minimum.

**Comment by Rangi42** [comment]
I just changed `_WS EQUS "\1"` to `_WS EQUS \1`, and now it works with both rgbds `master` and `new-lexer-electric-boogaloo`. Though I understand the new lexer is still in progress and may require more changes to the macro.

**Comment by Rangi42** [comment]
Also, I could get rid of the awkward multi-arg comma handling in one of two ways:

- `charmap "，", 235` (fullwidth comma or some other Unicode character)
- `text "Hello\, world!"` (this escape works with the old and new lexers)

Then the macro would be simplified to this:

```
text: macro
_arg = 1
    rept _NARG
_WS EQUS \1
_len = STRLEN("{_WS}")
_i = 1
        rept _len
          dw STRSUB("{_WS}", _i, 1)
_i = _i + 1
        endr
        PURGE _WS
        shift
_arg = _arg + 1
    endr
endm
```

And `text "Hello\, to", " the world!"` would do just what we want `dw "Hello\, to", " the world!"` to do in the first place. (Note that `db "\,"` is identical to `db ","`; the escape is pointless but harmless.)
